pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -1574,17 +1561,23 @@ impl RemoteTimelineClient {
     /// In-progress operations will still be running after this function returns.
     /// Use `task_mgr::shutdown_tasks(None, Some(self.tenant_id), Some(timeline_id))`
     /// to wait for them to complete, after calling this function.
-    pub(crate) fn stop(&self) -> Result<(), StopError> {
+    pub(crate) fn stop(&self) {
         // Whichever *task* for this RemoteTimelineClient grabs the mutex first will transition the queue
         // into stopped state, thereby dropping all off the queued *ops* which haven't become *tasks* yet.
         // The other *tasks* will come here and observe an already shut down queue and hence simply wrap up their business.
         let mut guard = self.upload_queue.lock().unwrap();
-        match &mut *guard {
-            UploadQueue::Uninitialized => Err(StopError::QueueUninitialized),
+        self.stop_impl(&mut guard);
+    }
+
+    fn stop_impl(&self, guard: &mut std::sync::MutexGuard<UploadQueue>) {
+        match &mut **guard {
+            UploadQueue::Uninitialized => {
+                info!("UploadQueue is in state Uninitialized, nothing to do");
```

Comment:
Do we need to `info!` this? Are you thinking of catching previously problematic cases with this while developing?

Reply:
Ah well, it seems we have an `info!` for all cases.

Reply:
No, put it there because I thought this would never run.

I _think_ the only place where we call stop()/shutdown() on Uninitialized state AND weren't making any noise previously is

https://github.com/neondatabase/neon/pull/7234/files#diff-74152cb7b3681c454c4156d530ce544ee8f15048c9f3cacc736dba5dc7c0d664L1285-L1290

and possibly

https://github.com/neondatabase/neon/pull/7234/files#diff-bfd1b153b1f900ef0cbd50a1dc308bf5ba812176e5ad65c8ae2a245bf0f12f84L2144

Can you think find any other places?

Do you think it's a frequent occurrence that we call on Uninitialized in those places?

If it's not a frequent occurrence, I think it's better to log than stay quiet, just in case there's a bug and we need to rely on logs to debug things.